juju/juju-controller
# [JUJU-5140] Write controller configuration file

Introduction of the configuration file as per [specification](https://docs.google.com/document/d/1JtS0MsmwR9sWABnmyLyCPo8M7qsEdlY7tKQfN89FLbk)

This patch ensures:
- that upon installation, the configuration file exists at the specified path and;
- when the `dbcluster` relation data for the application has a changed value for `db-bind-addresses, the new value is written to the configuration file.

To verify:
- `charmcraft pack` this charm.
- Bootstrap a new LXD controller using `--controller-charm-path <path-to-packed-artefact>`.
- `juju enable-ha`.
- Once all units have joined the relation, ssh to the controller machine and check _/var/lib/juju/agents/controller-0/agent.conf_. It should be populated along the lines of:
```
db-bind-addresses:
  controller/0: 10.246.27.181
  controller/1: 10.246.27.53
  controller/2: 10.246.27.12
```
